herd/AArch64Sem.ml

```diff
@@ -257,14 +283,26 @@ module Make
             do_write_mem sz an a v ii)
           ma ii
 
-      let ldr sz rd rs kr ii =
+      let ldr sz rd rs kr s ii =
         lift_memop
           (fun ma -> ma >>= fun a ->
            old_do_read_mem sz AArch64.N a ii >>= fun v ->
            write_reg rd v ii )
-          (get_ea rs kr ii) ii
+          (get_ea rs kr s ii) ii
+
+      (* Post-Indexed load immediate *)
+      and ldr_p sz rd rs k ii =
+        assert (k >= -256 && k <= 255);
+        let open AArch64Base in
+          (read_reg_ord rs ii)
+          >>= M.address_of >>= fun a1 -> M.deref a1
+          >>= fun a -> read_mem sz rd a ii
+          >>= (fun _ -> M.add a1 (V.intToV k))
+          >>= (fun v -> write_reg rs v ii)
```

Comment:
ah that's a good spot thank you, I should've seen that. 

I have fixed this by using the `>>|` operator to parallelize the computation to `read_mem` using `a` but also updating `a1`. I suspect this might be closer to the true intent of the instruction although it'd be worth a look from both you and @jalglave :)